pkg/stacks/templates/engines/helm2.go

```diff
@@ -0,0 +1,262 @@
+		her.Log.V(0).Info("Spec not found on claim; not creating engine configuration", "claim", claim)
+	}
+
+	her.Log.V(0).Info("Converting configuration", "spec", s)
+	configContents, err := yaml.Marshal(s)
+
+	her.Log.V(0).Info("Configuration contents as yaml", "configContents", configContents)
+
+	if err != nil {
+		her.Log.Error(err, "Error marshaling claim spec as yaml!", "claim", claim)
+		return nil, err
+	}
+
+	// Underneath, the yamler uses https://godoc.org/encoding/json#Marshal,
+	// which means that the bytes are UTF-8 encoded
+	// Theoretically we could get better performance by using a binary config
+	// map, but having a string makes it better for humans who may want to observe
+	// or troubleshoot behavior.
+	stringConfigContents := string(configContents)
+
+	// TODO get the engine type from the configuration
+	engineType := hc.Engine.Type
+
+	// TODO engine type should have a bit more structure;
+	// probably better to use an enum type pattern, with an
+	// engine name and its corresponding configuration file
+	// name in the same object
+	configKeyName := ""
+
+	if engineType == "helm2" {
+		configKeyName = "values.yaml"
+	}
+
+	configName := string(claim.GetUID())
+	generatedMap, err := generateConfigMap(configName, configKeyName, stringConfigContents, her.Log)
+
+	if err != nil {
+		her.Log.V(0).Info("Error generating config map!", "claim", claim, "error", err)
+		return nil, err
+	}
+
+	generatedMap.SetNamespace(claim.GetNamespace())
+
+	her.Log.V(0).Info("Generated config map to pass engine configuration", "configMap", generatedMap)
+
+	return generatedMap, err
+}
+
+// RunEngine executes the helm2 engine to create some resources.
+// TODO we could potentially have a method create the job, and a higher-level one execute it.
+func (her *Helm2EngineRunner) RunEngine(ctx context.Context, client client.Client, claim *unstructured.Unstructured, config *corev1.ConfigMap, stackSource string, hc *v1alpha1.HookConfiguration) (*unstructured.Unstructured, error) {
+	// TODO if there is no config specified, either use an empty config or don't specify
+	// one at all.
+
+	// TODO if we change this to meta.AsController, and we have the controller-runtime controller configured
+	// to Own Jobs, then we'll get a reconcile call when Jobs finish. However, we'd need to change the logic
+	// for the reconcile a bit to support that effectively. For example:
+	// - We wouldn't want to create jobs every time reconcile is run
+	//   * This means keeping track of created jobs somewhere and could also mean using deterministic job names
+	ownerRef := meta.AsOwner(meta.ReferenceTo(claim, claim.GroupVersionKind()))
+	var jobBackoff int32
+
+	// TODO target stack image will come from the stack object, or maybe the stack install object.
+	// Then for each resource behavior hook, we want to run the hook
+	// TODO update this to use the most recent format, where a hook is a structured object
+
+	resourceDir := fmt.Sprintf("/.registry/%s", hc.Directory)
```

Comment:
`.registry` we have a constant for that.  installjob.registryDirName

Maybe this should get moved to stacks.RegistryDirName for reuse?